tools/rtpsrelay/RelayHandler.cpp

```diff
@@ -116,7 +117,9 @@ int RelayHandler::handle_input(ACE_HANDLE)
     ACE_ERROR((LM_ERROR, "(%P|%t) %N:%l ERROR: RelayHandler::handle_input failed to deserialize RTPS header\n"));
     return 0;
   }
-  const auto empty_message = buffer->length() == 0;
+
+  // Beacon messages will be considered empty
```

Comment:
True, but the code doesn't do this.  The code below says that every message that happens to be 4 bytes long must be considered to be "empty" / "beacon".  We can afford a 4-byte memcmp to check.